lib/rules/max-params.js

```diff
@@ -79,7 +90,19 @@ module.exports = {
 		 * @private
 		 */
 		function checkFunction(node) {
-			if (node.params.length > numParams) {
+			const hasVoidThisParam =
+				node.params.length > 0 &&
+				node.params[0].type === "Identifier" &&
+				node.params[0].name === "this" &&
```

Comment:
[Testing] Removing this line doesn't break any tests. Which is a symptom of a bug, I think: that having a non-`this` first param typed as `void` isn't tested.

```ts
function test(a: void, b);
```

Reply:
Added in https://github.com/eslint/eslint/pull/19557/commits/8fbc807a76907be1e7c7562256dc1b973ac68c1d